src/iterators.rs

```diff
@@ -66,15 +65,15 @@ impl<'a, T> Iterator for AncestorIds<'a, T> {
     type Item = &'a NodeId;
 
     fn next(&mut self) -> Option<&'a NodeId> {
-        if let Some(current_id) = self.node_id.clone() {
```

Comment:
Same here.

Reply:
Looks good here as well!

Same comment as above about `self.node_id` never getting set to `None`.